security-framework/src/item.rs

```diff
@@ -762,6 +917,20 @@ pub fn add_item(add_params: CFDictionary) -> Result<()> {
     cvt(unsafe { SecItemAdd(add_params.as_concrete_TypeRef(), std::ptr::null_mut()) })
 }
 
+/// Translates to `SecItemUpdate`. Use `ItemSearchOptions` to build a `search_params` `CFDictionary`
+/// and `ItemUpdateOptions` to build an `update_params` `CFDictionary`.
+pub fn update_item(search_params: CFDictionary, update_params: CFDictionary) -> Result<()> {
```

Comment:
Should I also bring `add_item` into `ItemAddOptions::add_item` and deprecate it?

Reply:
Don't worry about add_item, unless you want to. I don't want to throw unrelated refactorings at you. 

Here take `&ItemSearchOptions` and `&ItemUpdateOptions`

Reply:
I figured I may as well deprecate `add_item` and create an `ItemAddOptions::add` method. Also an `ItemSearchOptions::delete` instead of `delete_item`.